### ISSUE TITLE
Solving non-trivial ODE with ChaosPy (SEIR Model)

### ISSUE THREAD

**Issue by Novermars** [jonathf/chaospy#318]
**Describe your problem**
I'm trying to do something similar to this [Julia Jupyter notebook](https://nbviewer.jupyter.org/github/djsegal/SimplePlots.jl/blob/master/binder/covid-19.ipynb), but using non-intrusive PCE using chaospy instead of Monte Carlo simulation. I've looked through the examples, but don't see how I can do this, as the examples all consider the solution of the basic linear differential equation.

So in short, given some (system of) ODE(s) which has some uncertain parameters and for which intrusive PCE is far from ideal, how can we use chaospy to solve/estimate such models?

**Additional context**
If I make it work, I have no problem writing it out as a full example for the tutorial, so other people can benefit from it!

**Comment by jonathf** [comment]
That shouldn't be to difficult.

If you by any chance are interested in an intrusive approach, I have a tutorial for that [here](https://chaospy.readthedocs.io/en/master/tutorials/intrusive_galerkin.html).
At the very least `scipy.odint` looks very similar to what the Julia code does with `ODEProblem` and `solve`.

Using an non-intrusive approach, you can either go for [point collocation](https://chaospy.readthedocs.io/en/master/tutorials/point_collocation.html) or [pseudo-spectral](https://chaospy.readthedocs.io/en/master/tutorials/pseudo_spectral_projection.html).
Following the tutorials, you should be able to do what you need by doing the following alterations:
* The `exponential_model` will need to be replaced with ODE-solver code.
* You parameters `alpha`, `beta`, `gamma` need to have a distribution assigned to them. E.g.
   ```python
   alpha = chaospy.Normal(0.2, 0.05)
   beta = chaospy.Normal(hyper_beta, 0.05)
   gamma = chaospy.Normal(0.5, 0.05)
   distribution = chaospy.J(alpha, beta, gamma)
   ```
* `coordinates` have to be created with step size. E.g. `coordinates = numpy.linspace(0, 200, 1000)`.

Let me know how it works out. :)

As new tutorials go, if you can make a nice looking example, I would be happy to incorporate it into the docs.

**Comment by Novermars** [comment]
Yes, thank you! I was struggling how to change the exponential_model to the ODE, but now it all makes sense.

The result can be found at: https://github.com/Novermars/polynomial_chaos_expansion/blob/master/SEIR_model/seir_model_pce.ipynb If you're interested in adding a more worked out example to the tutorials then let me know, shouldn't be time consuming and will probably help other people.

Am I correct to notice that all operations are single threaded?

**Comment by jonathf** [comment]
Looks good.
Let us make the notebook into a PR and discuss the details there. Do you need help setting it up?

Yes, the operations are single threaded. The bottleneck should be the evaluation though. Just update step 20 in your notebook to include something simple like [`multiprocess.Pool.map`](https://docs.python.org/3/library/multiprocessing.html) and the "slow" part of the code should become much faster.

**Comment by jonathf** [comment]
Implemented in #319.
